pkg/controllers/disruption/orchestration/queue.go

```diff
@@ -283,26 +284,24 @@ func (q *Queue) waitOrTerminate(ctx context.Context, cmd *Command) error {
 	// All replacements have been provisioned.
 	// All we need to do now is get a successful delete call for each node claim,
 	// then the termination controller will handle the eventual deletion of the nodes.
-	var multiErr error
-	for i := range cmd.candidates {
-		candidate := cmd.candidates[i]
-		q.recorder.Publish(disruptionevents.Terminating(candidate.Node, candidate.NodeClaim, cmd.Reason())...)
-		if err := q.kubeClient.Delete(ctx, candidate.NodeClaim); err != nil {
-			multiErr = multierr.Append(multiErr, client.IgnoreNotFound(err))
-		} else {
-			metrics.NodeClaimsDisruptedTotal.Inc(map[string]string{
-				metrics.ReasonLabel:       pretty.ToSnakeCase(string(cmd.reason)),
-				metrics.NodePoolLabel:     cmd.candidates[i].NodeClaim.Labels[v1.NodePoolLabelKey],
-				metrics.CapacityTypeLabel: cmd.candidates[i].NodeClaim.Labels[v1.CapacityTypeLabelKey],
-			})
+	errs := make([]error, len(cmd.candidates))
+	workqueue.ParallelizeUntil(ctx, len(cmd.candidates), len(cmd.candidates), func(i int) {
+		q.recorder.Publish(disruptionevents.Terminating(cmd.candidates[i].Node, cmd.candidates[i].NodeClaim, cmd.Reason())...)
```

Comment:
Not specific to this PR, but shouldn't we publish this event after we successfully set the deletion timestamp on the node?